### ISSUE TITLE
Invalid Sentry DSN message not logged for expected case

### ISSUE THREAD

**Issue by mathieuboudreau** [neuropoly/spinalcordtoolbox#1910]
### Prerequisites

* [x] Did you check past issues for similar issue(s) ?
* [ ] Can you reproduce the issue with any of the files found in `sct_testing_data` ?
  * Is this meant to be sct_**example**_data? If so, yes.
* [x] If you are using a development environment, is your environment updated ?

### Description

If an invalid DSN is setup, a message is supposed to be logged per [here](https://github.com/neuropoly/spinalcordtoolbox/blob/494325b65eb77733b0427178cdfa8f550a5822ae/scripts/sct_utils.py#L222-L224), but the exception doesn't appear to be caught to execute the logging statement for a test case described below.

### Steps to Reproduce

The e.g. cases are the steps that I did to force this bug locally.

1. Corrupt DSN (e.g. add/remove a single number in the DSN address of the `export SENTRY_DSN=`... line of *${SCT_DIR}/bin/sct_launcher*.
2. Create a bug (e.g. Added the statement `1/0` just after the line `sct.init_sct()
` in \_\_main\_\_ of sct_propseg.py
3. (Optional*) Add a print message in the first line of [` except raven.exceptions.InvalidDsn:`](https://github.com/neuropoly/spinalcordtoolbox/blob/494325b65eb77733b0427178cdfa8f550a5822ae/scripts/sct_utils.py#L222-L224) (e.g. `print('Sentry DSN not valid anymore, not reporting errors')`
   * *I couldn't find where the log.debug() statements are actually saved, which is why I added this hack.
4. Run function for case with bug (e.g. run batch_processing.sh)

**Expected behavior:**

The message `'Sentry DSN not valid anymore, not reporting errors'` should be printed, because of the code block below (and my hack in number no. 3 above):

https://github.com/neuropoly/spinalcordtoolbox/blob/494325b65eb77733b0427178cdfa8f550a5822ae/scripts/sct_utils.py#L222-L224

**Actual behavior:**

That message was not printed, see output below. Only `No handlers could be found for logger "sentry.errors"` was displayed, which appears to be sentry's default message for this case.

```
./batch_processing.sh 
Started at: 06.07.2018_

--
Spinal Cord Toolbox (master/494325b65eb77733b0427178cdfa8f550a5822ae)
Running /Users/mathieuboudreau/neuropoly/github/sct/scripts/sct_download_data.py -d sct_example_data

Trying URL: https://osf.io/kjcgs/?action=download
Downloading 20180525_sct_example_data.zip...
Status: 100%|###################################################################################| 44.3M/44.3M [00:02<00:00, 14.9MB/s]

Check if folder already exists...
WARNING: Folder sct_example_data already exists. Removing it...
rm -rf sct_example_data

Unzip data to: /Users/mathieuboudreau/neuropoly/github/sct

Remove temporary file...
Done!

Total processing time: 0 min 6 s


--
Spinal Cord Toolbox (master/494325b65eb77733b0427178cdfa8f550a5822ae)
Running /Users/mathieuboudreau/neuropoly/github/sct/scripts/sct_propseg.py -i t2.nii.gz -c t2
Traceback (most recent call last):
  File "/Users/mathieuboudreau/neuropoly/github/sct/scripts/sct_propseg.py", line 343, in <module>
    1/0
ZeroDivisionError: integer division or modulo by zero
Sentry is attempting to send 1 pending error messages
Waiting up to 10 seconds
Press Ctrl-C to quit
No handlers could be found for logger "sentry.errors"
Note: you can opt out of Sentry reporting by editing the file ${SCT_DIR}/bin/sct_launcher and delete the line starting with "export SENTRY_DSN"
Total processing time: 0 min 2 s
```

**Reproduces how often:**

100% of the times I tested.

### State of spinalcordtoolbox

```
 ./batch_processing.sh 
Started at: 06.07.2018_

--
Spinal Cord Toolbox (master/494325b65eb77733b0427178cdfa8f550a5822ae)
Running /Users/mathieuboudreau/neuropoly/github/sct/scripts/sct_download_data.py -d sct_example_data
```

### Additional Information

I saw @zougloub's message on slack about disabling Sentry for dev/testing. (I'm not "alioth"). I haven't and don't intend to test beyond this case using a valid DSN, I just fell on this while trying to port SCT's Sentry setup to AxonDeepSeg.

**Comment by mathieuboudreau** [comment]
I was actually able to get that statement to execute/print (testing with my own account's DSN), however it's a somewhat subtle case.

There are two keys in the Sentry settings, "DSN" and "DSN (Deprecated)" (which contains the secret key). It seems like you are using a setup for the "DSN (Deprecated)". Using my own "DSN (Deprecated)", I don't get the print message. I don't even get the logged/printed message for this case if I add/remove a letter/number in the key, or if I regenerate a new one and delete the key that I set it up for.

I do however get that error message to display if I use my own *valid* "DSN" key, possibly because you're using an older version of Sentry/SDK which requires the secret?

This issue could likely be closed, I just thought I'd let you know of this odd behaviour – I thought that error would be logged for any invalid DSN, or at least expired/deleted ones as the comment above the logging statement suggested. Seems like it only actually logs if the secret key is missing, regardless of if the public/secret key the valid or expired (like the comment `# This could happen if sct staff change the dsn` suggested).

**Comment by jcohenadad** [comment]
Thanks a lot for those investigations @mathieuboudreau! 
Maybe we should switch to the new DSN?

**Comment by mathieuboudreau** [comment]
@jcohenadad from all I've read in the Sentry github issues, the Sentry will not accept the private DSN that SCT is currently using "soon", but it's not clear when that will be. So yes, it's likely a good idea to switch from the private DSN we're currently using to the public one available on the Sentry settings page.

One thing to note is that the current version of raven-python that sct is using (6.2.1) isn't compatible with the public-only DSN; [they fixed that bug recently in version 6.8.0](https://github.com/getsentry/raven-python/commit/3f9f93137b09bd0e510709f198fef0981ce83b7c). So if you want to switch to using the public DSN, you'll have to change the private DSN to the public one on the NeuroPoly server (in the "stats_dsn.txt" file), and raise the version requirement to at least 6.9.0 in these two locations: 

https://github.com/neuropoly/spinalcordtoolbox/blob/0161a53eff5bc5b90ee4c8dcee2586cfde276cb9/install/requirements/requirementsSetup.txt#L11

https://github.com/neuropoly/spinalcordtoolbox/blob/0161a53eff5bc5b90ee4c8dcee2586cfde276cb9/install/requirements/requirementsPip.txt#L6

I tested it using the my own public DSN with SCT + raven==6.8.0, and it worked (Sentry received report for the fake bug I created).
